### DISCUSSION TITLE
Routing: handle passed via-points more gracefully

### DISCUSSION THREAD

**Discussion by xax** [organicmaps/organicmaps#8035] [Ideas]
Consider the situation when you want to modify a calculated from-to route by adding via-points (because you know better or you want to force navigation to guide you along a self-chosen refined route).

If you happen to be imprecise with your finger adding a via, you end up adding a via slightly off road or on the opposite direction lane of a multi-lane road.

In that case, Organic Maps *insists* on you passing **through** the via-point and kind-of stubbornly does not get along with it (and with you) targeting the next via (or end point) , even as you clearly follow the route between the supposedly "*missed*" via and the next via or end point. OM navigation should detect that and consider the "*near miss*" a passed via – at least after a threshold of ignored nav instructions back to the supposedly missed via.

(E.g. on a motorbike, using OM for acoustic navigation only, it is a major disruption having to delete the via manually after such "near miss".)

**Comment by pm4rcin** [comment]
I think you request is a duplicate of https://github.com/organicmaps/organicmaps/issues/2699.
